enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/ServiceDiscoveryEnricher.java

```diff
@@ -81,79 +79,68 @@ public ServiceDiscoveryEnricher(EnricherContext buildContext) {
-                    annotations.put(PREFIX + "/" + PATH             , getConfig(Config.path, path));
-                }
-                annotations.put(PREFIX + "/" + PORT           , getConfig(Config.port, port));
-                if (getConfig(Config.descriptionPath, descriptionPath) != null) {
-                    annotations.put(PREFIX + "/" + DESCRIPTION_PATH , getConfig(Config.descriptionPath, descriptionPath));
-                }
-                for (String annotationName : annotations.keySet()) {
-                    log.info("Add %s annotation: \"%s\" : \"%s\"", PREFIX, 
-                            annotationName, annotations.get(annotationName));
-                }
-                return annotations;
+            @Override
+            public void visit(ServiceBuilder serviceBuilder) {
+                addAnnotations(serviceBuilder);
             }
-        }
 
-        return super.getAnnotations(kind);
+        });
+        
+        
     }
 
-    @Override
-    public Map<String, String> getLabels(Kind kind) {
-
-        if (kind == Kind.SERVICE) {
-
-            if (!tried) {
-                tryCamelDSLProject();
-                //try more project types in future, last match wins
-                tried = true;
-            }
+    protected void addAnnotations(ServiceBuilder serviceBuilder) {
 
-            if (discoverable != null) {
-                Map<String, String> labels = new HashMap<>();
-                String labelName = PREFIX;
-                labels.put(labelName, getConfig(Config.discoverable, discoverable));
-                log.info("Add %s label: \"%s\" : \"%s\"", PREFIX, 
-                        labelName, labels.get(labelName));
-                return labels;
+        if (serviceBuilder.buildSpec() != null) {
+            List<ServicePort> ports = serviceBuilder.buildSpec().getPorts();
+            if (! ports.isEmpty()) {
+                ServicePort firstServicePort = ports.iterator().next();
+                port = firstServicePort.getPort().toString();
+                log.info("Using first mentioned service port '%s' " , port);
+            } else {
+                log.warn("No service port was found");
             }
         }
-        return super.getLabels(kind);
-    }
-
-    @Override
-    public void addMissingResources(final KubernetesListBuilder listBuilder) {
-        listBuilder.accept(new TypedVisitor<ServiceBuilder>() {
-
-            @Override
-            public void visit(ServiceBuilder serviceBuilder) {
-                
-                List<ServicePort> ports = serviceBuilder.buildSpec().getPorts();
-                if (! ports.isEmpty()) {
-                    ServicePort firstServicePort = ports.iterator().next();
-                    port = firstServicePort.getPort().toString();
-                    log.info("Using first mentioned service port '%s' " , port);
+        
+        tryCamelDSLProject();
+        
+        if (discoverable != null) {
```

Comment:
Sorry if I get it wrong, but is it know that you *have* set `discoverable` config to `true` otherwise the enricher would do nothing ? Why not the other way round and use a 'hidden' config which much be set to *not* use this enricher when its applied ?

Not sure that I understand the idea of the `discoverable` config parameter.

Reply:
If the project is not one of the ones we recognize (i.e. not RestDSL) then we won't add the annotations. If it's set to 'false' in the config then the annotations are still added but the label is set to "false". This may come in handy is you want to expose later but not right now.